lib/pengine/native.c

```diff
@@ -824,35 +818,38 @@ pe__common_output_text(pcmk__output_t *out, resource_t * rsc, const char *pre_te
     }
 
     CRM_LOG_ASSERT(offset > 0);
-    if(flagOffset > 0) {
-        fprintf(out->dest, "%s (%s)%s%s", buffer, flagBuffer, desc?" ":"", desc?desc:"");
-    } else {
-        fprintf(out->dest, "%s%s%s", buffer, desc?" ":"", desc?desc:"");
-    }
 
-    if ((options & pe_print_rsconly)) {
-        if (options & pe_print_suppres_nl) {
-            /* nothing */
-        } else {
-            fprintf(out->dest, "\n");
-        }
-    } else if (g_list_length(rsc->running_on) > 1) {
+    if (!(options & pe_print_rsconly) && g_list_length(rsc->running_on) > 1) {
         GListPtr gIter = rsc->running_on;
-
-        fprintf(out->dest, "[");
+        gchar **arr = calloc(g_list_length(rsc->running_on)+1, sizeof(gchar *));
+        int i = 0;
+        char *total = NULL;
 
         for (; gIter != NULL; gIter = gIter->next) {
             node_t *n = (node_t *) gIter->data;
-            fprintf(out->dest, " %s", n->details->uname);
-        }
-        fprintf(out->dest, " ]");
-        if (options & pe_print_suppres_nl) {
-            /* nothing */
-        } else {
-            fprintf(out->dest, "\n");
+            arr[i] = (gchar *) n->details->uname;
+            i++;
         }
+
+        total = g_strjoinv(" ", arr);
+        unames = crm_strdup_printf(" [ %s ]", total);
+
+        free(total);
```

Comment:
And here's an example of why we try to avoid mixing and matching memory allocation :)

Reply:
Switched to using g_strfreev for the moment.  If we end up changing how the list is made, I'll change how it's freed too.

Reply:
g_strjoinv() needs g_free()